### PR TITLE
feature/alk-resource-handling

### PR DESCRIPTION
**Pull request by dcepil**
added support for `!alk add <resource>`:
- checks if the calling user has the `whitelisted` field in the database set to `true` and whether or not the resource already exists (judging by the previously checked resource link) - if the user is `whitelisted` and the resource does not yet exist, adds the resource to the database for further moderator approval
- updated models and migrations for `user` and `media`, connected to this implementation
- added [`mime-types`](https://www.npmjs.com/package/mime-types) as a dependency to support validating the passed resource

### CODE REVIEW

File: commands/alk.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+const checkResource = (resource) => {
+  if (checkURL(resource)) return mime.lookup(resource);
+};
+
+const addResource = async (msg, args) => {
+  const user = await userModel.findOrCreateByDiscordId(msg.author.id);
+  const passedResource = args[1];
+  const resourceExists = await mediaModel.findOneByMediaContent(passedResource);
+  if (user.whitelisted && checkResource(passedResource) && !resourceExists) {
```
Comment (VuongRon) [inline commands/alk.js]: I would do the `user.whitelisted` check on line 21 so we don't have to wait for the media query just to stop processing.

File: commands/alk.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return;
+      });
+    const message = "Successfully added the resource for approval.";
+    return embedService.embedMessage(msg, args, message);
+  }
+};
+
+const getResource = (msg, args) => {
+  
+};
```
Comment (VuongRon) [inline commands/alk.js]: alkNSNDC

**Review by VuongRon — APPROVED** [review]
🏃

File: commands/alk.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    return mime.lookup(resource);
+  }
+};
+
+const addResource = async (msg, args) => {
+  const user = await userModel.findOrCreateByDiscordId(msg.author.id);
+  const passedResource = args[1];
+  const resourceExists = await mediaModel.findOneByMediaContent(passedResource);
+  const resourceValid = await checkResource(passedResource);
+  if (user.whitelisted && resourceValid && !resourceExists) {
```
Comment (VuongRon) [inline commands/alk.js]: Move `user.whitelisted` up a few lines so we don't have to await the other calls

Comment (dcepil) [reply commands/alk.js]: Fixed.

**Review by VuongRon — APPROVED** [review]
🏃 💨

**Pull request merged**